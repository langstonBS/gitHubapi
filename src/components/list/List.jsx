import React from 'react';
import PropTypes from 'prop-types';

const List = ({ repos }) => {
  const listElements = repos.map(repo => (
    <li key={repo.name}>
      <a href= {repo.html_url}> {repo.name} </a>
    </li>
  ));

  return (
    <ul>
      {listElements}
    </ul>
  );
};

List.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default List;
  
