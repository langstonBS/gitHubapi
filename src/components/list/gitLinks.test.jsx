import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(< List repos={[{ name:'myName', html_url:'url' }, { name:'myName', html_url:'url' }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
