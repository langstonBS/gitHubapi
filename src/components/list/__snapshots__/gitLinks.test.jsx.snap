// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<ul>
  <li
    key="myName"
  >
    <a
      href="url"
    >
       
      myName
       
    </a>
  </li>
  <li
    key="myName"
  >
    <a
      href="url"
    >
       
      myName
       
    </a>
  </li>
</ul>
`;
